inject.js

```diff
@@ -415,27 +439,46 @@
         // Process the DOM nodes lazily
         requestIdleCallback(_ => {
           mutations.forEach(function(mutation) {
-            forEach.call(mutation.addedNodes, function(node) {
-              if (typeof node === "function")
-                return;
-              checkForVideo(node, node.parentNode || mutation.target, true);
-            });
-            forEach.call(mutation.removedNodes, function(node) {
-              if (typeof node === "function")
-                return;
-              checkForVideo(node, node.parentNode || mutation.target, false);
-            });
+            switch (mutation.type) {
+              case 'childList':
+                forEach.call(mutation.addedNodes, function(node) {
+                  if (typeof node === "function")
+                    return;
+                  checkForVideo(node, node.parentNode || mutation.target, true);
+                });
+                forEach.call(mutation.removedNodes, function(node) {
+                  if (typeof node === "function")
+                    return;
+                  checkForVideo(node, node.parentNode || mutation.target, false);
+                });
+                break;
+              case 'attributes':
+                if ((mutation.target.tagName == 'APPLE-TV-PLUS-PLAYER') && (mutation.target.attributes['aria-hidden'].value == "false")) {
+                  var flattenedNodes = getShadow(document.body)
+                  var node = flattenedNodes.filter(x => x.tagName == 'VIDEO')[0]
+                  var oldController = flattenedNodes.filter(x => x.classList.contains('vsc-controller'))[0]
+                  if (oldController) {
+                    oldController.remove()
+                  }
+                  checkForVideo(node, node.parentNode || mutation.target, true);
+                }
+                break;
+            };
           });
         }, {timeout: 1000});
       });
-      observer.observe(document, { childList: true, subtree: true });
+      observer.observe(document, {
+        attributeFilter: ['aria-hidden'],
```

Comment:
I was able to figure this one out too based on your prior comments...

I definitely see the challenge of the video not coming into existence until the element is shown. That said, I'm not sure we would ever want to scan/initialize hidden elements anyway, so this may make sense to keep in.

Reply:
i think that each year, the shadow dom specification changes, so it will just be a game of cat and mouse for the rest of eternity

Reply:
I agree. I don't think this is specific to the shadow dom though, I think it's more of a single page application design choice. With the popularity of react and similar single page applications, I can see how it may become a game of cat and mouse though as you say lol